engine/api/workflow_run.go

```diff
@@ -161,6 +161,26 @@ func (api *API) getWorkflowRunsHandler() service.Handler {
 	}
 }
 
+func (api *API) deleteWorkflowRunsBranchHandler() service.Handler {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		vars := mux.Vars(r)
+		key := vars["key"]
+		name := vars["permWorkflowName"]
+		branch := vars["branch"]
+
+		wfIDs, err := workflow.LoadRunsIDByTags(api.mustDB(), key, name, map[string]string{"git.branch": branch})
+		if err != nil {
+			return err
+		}
+
+		if err := workflow.MarkWorkflowRunsAsDelete(api.mustDB(), wfIDs); err != nil {
```

Comment:
Do we have to stop them all?

Reply:
No, with future commit it will be deleted 24h after, so it will be stopped